dags/copy_deduplicate.py

```diff
@@ -115,6 +115,71 @@
 
     copy_deduplicate_all >> impression_stats_flat
 
+    # Derived tables for messaging-system.
+
+    # CFR
+    messaging_system_cfr_users_daily = bigquery_etl_query(
+        task_id='messaging_system_cfr_users_daily',
+        project_id='moz-fx-data-shared-prod',
+        destination_table='cfr_users_daily_v1',
+        dataset_id='messaging_system_derived',
+        email=['najiang@mozilla.com'],
+    )
+
+    messaging_system_cfr_users_last_seen = bigquery_etl_query(
+        task_id='messaging_system_cfr_users_last_seen',
+        project_id='moz-fx-data-shared-prod',
+        destination_table='cfr_users_last_seen_v1',
+        dataset_id='messaging_system_derived',
+        email=['najiang@mozilla.com'],
+    )
```

Comment:
```suggestion
        depends_on_past=True,
    )
```

The last_seen queries all need to be marked as `depends_on_past`.

Reply:
Ah, good catch. Thanks! Always wonder why it's not the default behavior.

Reply:
> Always wonder why it's not the default behavior.

Because it's a pain ;) When depends_on_past is False, backfills can proceed in parallel and a failure one day doesn't have any impact on subsequent days. When depends_on_past is true, there's a certain amount of handholding that needs to happen when things aren't running smoothly. So it's nice to avoid it, but in the last_seen case it's necessary.